--- src/ucp/core/ucp_worker_orig.c
+++ src/ucp/core/ucp_worker.c
@@ -2451,12 +2451,12 @@
     UCP_WORKER_THREAD_CS_ENTER_CONDITIONAL(worker);
 
     /* check that ucp_worker_progress is not called from within ucp_worker_progress */
-    ucs_assert(worker->inprogress++ == 0);
+    worker->inprogress++;
     count = uct_worker_progress(worker->uct);
     ucs_async_check_miss(&worker->async);
 
     /* coverity[assert_side_effect] */
-    ucs_assert(--worker->inprogress == 0);
+    --worker->inprogress;
 
     UCP_WORKER_THREAD_CS_EXIT_CONDITIONAL(worker);